src/lib/CMakeLists.txt

```diff
@@ -0,0 +1,92 @@
+# Some operating systems don't like compiling a library with
+# no source files
+
+#add_library(openstudiolib_static STATIC empty.cpp)
+
+#target_link_libraries(
+# openstudiolib_static
+# PUBLIC
+# openstudio_utilities
+# openstudio_airflow
+# openstudio_model
+# openstudio_energyplus
+# openstudio_measure
+# openstudio_osversion
+# openstudio_sdd
+# openstudio_isomodel
+# openstudio_gbxml
+# openstudio_radiance
+#)
+#
+
+# Some operating systems don't like compiling a library with
+# no source files
+add_library(openstudiolib SHARED empty.cpp)
+
+
+target_link_libraries(
+ openstudiolib
+ PRIVATE
+ openstudio_utilities
+ openstudio_airflow
+ openstudio_model
+ openstudio_energyplus
+ openstudio_measure
+ openstudio_osversion
+ openstudio_sdd
+ openstudio_isomodel
+ openstudio_gbxml
+ openstudio_radiance
+)
+
+target_link_libraries(
+ openstudiolib
+ INTERFACE
+ CONAN_PKG::boost
+ CONAN_PKG::jsoncpp
+ CONAN_PKG::cpprestsdk
+ CONAN_PKG::openssl
+ CONAN_PKG::pugixml
+)
+
+target_compile_definitions(openstudiolib INTERFACE "-DSHARED_OS_LIBS")
+
+# We cannot make the libs that make up the shared library public dependency at all, if we do
+# then we have to export and install them as well, which we probably don't want to do.
+
+# If we try to include directories from the dependent libraries directly, we have to use
+# generator expressions to get the specific configuration's INTERFACE_INCLUDE_DIRECTORIES
+
+# This also does not seem to do what we ultimately want, as it would also require
+# that we export the .a files.
+
+# So here we are, I'm manually exporting the directories, it's probably a mistake
+# TODO reevaluate this.
```

Comment:
When?

Reply:
Well the only mistake i can detect is that many of the conan library dependencies are still exposed to any consuming projects, meaning they have to provide those same conan libs. Note for example that the only consuming project in existence (OS App) has a conan file that replicates this project's conan file. I think me and @lefticus discussed on the phone and it is a big maybe if this is required. It depends if all of the symbols in the conan dependencies that are exposed in the api headers are included in the openstudio.dll. It is hard to guarantee that so to be safe we expose the conan libs as interface libraries.